iree_tests/conftest.py

```diff
@@ -375,12 +375,17 @@ def runtest(self):
             raise e
 
     def test_compile(self):
-        proc = subprocess.run(self.compile_args, capture_output=True, cwd=self.test_cwd)
+        compile_env = os.environ.copy()
+        compile_env["SPEC_LOCATION"] = os.getenv("SPEC_LOCATION", default=self.test_cwd)
+        cmd = subprocess.list2cmdline(self.compile_args)
+        proc = subprocess.run(cmd, env=compile_env, shell=True, capture_output=True, cwd=self.test_cwd)
```

Comment:
We should be particularly careful about introducing "spooky action at a distance" (as Ben's words :P) here. When a test fails, the reproducer command that we log should be easy to replicate. Here's an example: https://github.com/iree-org/iree/actions/runs/8929912892/job/24559674205#step:9:909
`Invoked with: cd /home/esaimana/actions-runner/_work/iree/iree/SHARK-TestSuite/iree_tests/pytorch/models/sdxl-vae-decode-tank && iree-compile model.mlirbc --iree-hal-target-backends=vulkan-spirv -o model_gpu_vulkan_real_weights.vmfb`

What does the output look like with this new environment variable mapping?

I'm also thinking we should add the ability to log the compile and run commands _even for passing tests_. That way, we can point developers at the latest CI logs whenever they ask "what are the SDXL flags on main" and they'll be able to copy/paste from the location that ran the tests (even/especially if they passed). I think we can include logs for passing tests based on some changes in conftest.py and/or our flags/pytest.ini settings: https://docs.pytest.org/en/stable/how-to/logging.html.

Reply:
<img width="576" alt="image" src="https://github.com/nod-ai/SHARK-TestSuite/assets/77521230/49d621bc-dda6-47a5-95f9-24300bcbbea1">
Lol interesting output. SPEC_LOCATION isn't expanded so I guess not the easiest to reproduce. also any idea what's going on with the run output?

Reply:
Ah! Glad we checked :P
The previous code expected a list, but `subprocess.list2cmdline` generates a string:
https://github.com/nod-ai/SHARK-TestSuite/blob/9e9774632d32674e9ffa44b0a2ffc054f6c590ad/iree_tests/conftest.py#L448-L449

I'm also thinking we could inline the `--flagfile=` in the output (maybe log both the flagfile version and the expanded version)

Reply:
Ahh okay got it! Yeah it would be good to have the flagfile arg expanded. I can expand the env variable for these error outputs too

Reply:
<img width="603" alt="image" src="https://github.com/nod-ai/SHARK-TestSuite/assets/77521230/15e6e314-9086-43e4-a13e-34e77b2e56e5">

Thanks, looking like normal human readable output now :)